compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -1774,6 +1774,8 @@ bool OMR::Power::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode
       {
       case TR::vadd:
       case TR::vsub:
+         if (dt == TR::Int16 || dt == TR::Int8)
+            return true;
```

Comment:
You seem to be relying on this falling through to the next case when this condition isn't true, but IMO that seems like a potential source of confusion. I think it would be better to add the checks for `Int32`, `Float`, and `Double` here and avoid allowing fallthrough.

Reply:
Good point, fixed!